jupyter_server/serverapp.py

```diff
@@ -2377,7 +2377,11 @@ def local_url(self) -> str:
         parts = self._get_urlparts(include_token=True)
         # Update with custom pieces.
         if not self.sock:
-            parts = parts._replace(netloc=f"127.0.0.1:{self.port}")
+            if ":" in self.ip:
+                localhost = "[::1]"
+            else:
+                localhost = "127.0.0.1"
```

Comment:
```suggestion
            localhost = "[::1]" if ":" in self.ip else "127.0.0.1"
```
From linter:
`jupyter_server/serverapp.py:2380:13: SIM108 Use ternary operator `localhost = "[::1]" if ":" in self.ip else "127.0.0.1"` instead of `if`-`else`-block`